### PR TITLE
remove process substitution

### PR DESCRIPTION
**Pull request by Despire**
closes https://github.com/berops/claudie/issues/1126

The issue is when the combination `bash -c` `<()` is used while the entrypoint of the docker image is not a shell. Thus replace the proces subsitution `<()` by directly creating a temporary file.

for a quick verification of the issue one can try 

docker run -it --rm [docker.io/library/alpine:3.18.4](http://docker.io/library/alpine:3.18.4) tail -f /dev/null 

and in another terminal exec into the container and  execute a simple command like `bash -c 'echo <(echo 1)'` and look at the `ps` output, the subprocess won't be terminated

### CODE REVIEW

**Review by bernardhalas — APPROVED** [review]
I'm no Gopher, but just wondering if the copy/pasted code is worth delegating to a small function.

File: internal/kubectl/kubectl.go [hunk 1]
```diff
@@ -186,15 +384,36 @@ func (k Kubectl) runWithOutput(command string, options ...string) ([]byte, error
...
+		return "", nil, nil
+	}
+
+	id := uuid.New()
+	tmpName := fmt.Sprintf("/tmp/k%s", hex.EncodeToString(id[:]))
+	if err := os.WriteFile(tmpName, []byte(k.Kubeconfig), os.ModePerm); err != nil {
+		return "", nil, err
 	}
+
+	cleanup := func() error { return os.Remove(tmpName) }
```
Comment (cloudziu) [inline internal/kubectl/kubectl.go:416]:
```suggestion
cleanup := func() {
    if err := os.Remove(tmpName); err != nil {
        log.Err(err).Msg("failed to cleanup kubeconfig")
    }
}
```

And then do a shoret `defer`: 
```
if cleanup != nil {
		defer cleanup()
	}
```
WDYT?

**Review by cloudziu — APPROVED** [review]

File: internal/kubectl/kubectl.go (hunk 1 above)
Comment (bernardhalas) [reply internal/kubectl/kubectl.go:416]: Or even `defer cleanup(doCleanup)` so even the condition could be evaluated in the function itself.

**Review by cloudziu — APPROVED** [review]
LGTM 👍

**Review by bernardhalas — APPROVED** [review]
Thanks for the refactor @Despire

**Pull request merged**